internal/gensupport/resumable.go

```diff
@@ -241,13 +245,32 @@ func (rx *ResumableUpload) Upload(ctx context.Context) (resp *http.Response, err
 			default:
 			}
 
-			resp, err = rx.transferChunk(ctx)
+			var rCtx context.Context
+			var cancel context.CancelFunc
+
+			rCtx = ctx
+			if rx.ChunkTransferTimeout != 0 {
+				rCtx, cancel = context.WithTimeout(ctx, rx.ChunkTransferTimeout)
+			}
+
+			resp, err = rx.transferChunk(rCtx)
 
 			var status int
 			if resp != nil {
 				status = resp.StatusCode
 			}
 
+			// The upload should be retried if the rCtx is canceled due to a timeout.
+			select {
+			case <-rCtx.Done():
+				if rCtx.Err() == context.DeadlineExceeded {
+					// Cancel the context for rCtx
+					cancel()
```

Comment:
I think this would panic with chunkTransferTimeout = 0 because the function was never set to anything. Did you test this case?

Reply:
Yes I have done manual testing, it is not having panic. I have assign rCtx=ctx and override the rCtx value in case of transfertimeout non zero. If it is zero then rCtx=ctx.